packages/app/android/src/main/java/io/invertase/firebase/common/UniversalFirebaseModule.java

```diff
@@ -21,12 +21,19 @@
 
 import java.util.HashMap;
 import java.util.Map;
+import java.util.Set;
 import java.util.concurrent.ExecutorService;
 import java.util.concurrent.Executors;
+import java.util.concurrent.RejectedExecutionHandler;
+import java.util.concurrent.ThreadPoolExecutor;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.SynchronousQueue;
 
 import javax.annotation.OverridingMethodsMustInvokeSuper;
 
 public class UniversalFirebaseModule {
+  private static final int MAXIMUM_POOL_SIZE = 20;
+  private static final int KEEP_ALIVE_SECONDS = 3;
```

Comment:
How did you arrive at these constants?
- what happens if they are much smaller, or much larger?
- is there a possibility that with certain workloads these values could exhaust resources and cause a crash?
- is it possible that different workloads might benefit from different values?

I had a separate thought with regard to this implementation too - that is related - in the case there are some errors with regard to threading, even if the underlying implementation is "correct" (gets the intended executor for the intended firebase task pathway), is there a way to toggle off the multi-thread behavior?

I *think* if these were a tunable in firebase.json, especially the pool size, and it was set to 0, then that threadpool would always reject and everything would go single-thread? (though I'm not sure you can create a threadpool with a zero limit...)

Reply:
I think the biggest win by allowing tuning is to tune it to 0 (if that's possible?) such that everything spills to single thread executor and we then have a possible fallback behavior of "works just like it did before"

That work up front could stop a lot of possible frantic work later if our assumptions about multi-thread being okay in some cases is wrong :-)